hosts/web01/pgbackrest.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+let
+  # A "stanza" is pgBackrest speak for a postgresql cluster
+  stanza = "web01";
+
+  mkBackupUnits = { type, onCalendar }: {
+    systemd.timers."pgbackrest-${type}" = {
+      partOf = [ "pgbackrest-${type}.service" ];
+      wantedBy = [ "timers.target" ];
+      timerConfig = {
+        OnCalendar = onCalendar;
+        Unit = "pgbackrest-${type}.service";
+      };
+    };
+
+    systemd.services."pgbackrest-${type}" = {
+      serviceConfig.Type = "oneshot";
+      serviceConfig.ExecStart = "${lib.getExe pkgs.pgbackrest} --type=${type} --stanza=${stanza} backup";
+      serviceConfig.User = "postgres";
+    };
+  };
+
+in
+lib.mkMerge [
+  # Config
+  {
+    environment.etc."pgbackrest/pgbackrest.conf".text = lib.generators.toINI { } {
+      "${stanza}" = {
+        pg1-path = config.services.postgresql.dataDir;
+      };
+
+      global = {
+        repo1-retention-full = 3;
+        repo1-type = "cifs";
+        repo1-path = "/mnt/pgbackrest";
```

Comment:
We also have a hetzner storage box for each physical machine, in case you need this (comes with borgbackup/cifs/sftp support).